### PR TITLE
Remove keyspace info in metrics and some useless metrics info in schema 

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #7640

Problem Summary:

### What is changed and how it works?
Considering a large number of keyspace info may lead grafana stucks, so revert https://github.com/pingcap/tiflash/pull/7679 to remove keyspace info in metrics.

Considering without keyspace info in metrics, schema version is meaning less, so delete the useless metrics.
image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TiDB/Schema/SchemaBuilder.cpp [hunk 1]
```diff
@@ -895,50 +895,99 @@ void SchemaBuilder<Getter, NameMapper>::syncAllSchema()
...
                 }
             }
-        }
+        };
+        sync_all_schema_wait_group->schedule(task);
     }
+    sync_all_schema_wait_group->wait();
+
+
+    // for (const auto & db : all_schemas)
```
Comment (flowbehappy) [inline dbms/src/TiDB/Schema/SchemaBuilder.cpp:950]: Remove those useless code

**Review by flowbehappy — APPROVED** [review]
LGTM

**Review by lidezhu — APPROVED** [review]
LGTM

File: dbms/src/Common/TiFlashMetrics.h [hunk 2]
```diff
@@ -110,8 +108,6 @@ namespace DB
         F(type_passthrough_none_compression_remote, {"type", "passthrough_none_compression_remote"}),                                               \
         F(type_passthrough_lz4_compression, {"type", "passthrough_lz4_compression"}),                                                               \
         F(type_passthrough_zstd_compression, {"type", "passthrough_zstd_compression"}))                                                             \
-    M(tiflash_schema_version, "Current version of tiflash cached schema", Gauge)                                                                    \
-    M(tiflash_sync_schema_applying, "Whether the schema is applying or not (holding lock)", Gauge)                                                  \
```
Comment (JaySon-Huang) [inline dbms/src/Common/TiFlashMetrics.h:114]: Keeping this metrics maybe helpful for us locating bugs that the threads is stuck by schema sync

Comment (hongyunyan) [reply dbms/src/Common/TiFlashMetrics.h:114]: Fixed

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Pull request merged**